### PR TITLE
Feature: improve docs generation

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
I'd noticed that the vercel hosted doc previews were broken, this is work that aims to fix that as well as some other issues in working with the documentation.

image

- [ ] I've added a unit test to test for potential regressions of this bug. n/a
- [ ] The changelog has been updated, if applicable. n/a
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

See the commit message for details on what changed.

### CODE REVIEW

**Review by ajacksified — APPROVED** [review]

**Review by matthieubosquet — APPROVED** [review]
LGTM

**Comment by ThisIsMissEm** [comment]
I wasn't able to merge this yet because something is causing the source links to not work in vercel (they work locally though)

**Comment by matthieubosquet** [comment]
By the way @ThisIsMissEm , is it correct that the publish-website.yml workflow (triggering on any push to main) is removed without replacement?

**Comment by matthieubosquet** [comment]
Tagging @kay-kim as this will affect her docs generation workflow.

**Comment by ThisIsMissEm** [comment]
> By the way @ThisIsMissEm , is it correct that the publish-website.yml workflow (triggering on any push to main) is removed without replacement?

yes, it's not needed as it's automatically deployed to vercel anyway, there shouldn't be a need to increase the repo size with doc builds

**Comment by matthieubosquet** [comment]
I'm trying to double check if there is any intended audience for https://inrupt.github.io/solid-client-js/ or if we should just delete the gh-pages branch and kill gh-pages on this repo.

@ajacksified what do you think?
It could be a relatively cheap fallback, but if there's no action left to update it (publish-website), we should probably just kill it altogether, not good to have outdated stuff lingering.

File: docs/api/conf.py [hunk 1]
```diff
@@ -67,9 +67,9 @@
 #html_theme = 'alabaster'
 
 html_theme = 'inrupt'
-html_theme_path = ['./build/docs-assets/themes']
+html_theme_path = ['./docs-assets/themes']
 
-html_copy_source = False
+html_copy_source = True
```
Comment (kay-kim) [inline docs/api/conf.py:72]: Q: Curious, but why are we setting this option to True?

**Comment by matthieubosquet** [comment]
@kay-kim OK, thank you for the precision.

Then unless @ajacksified you have more information, I would suggest we nuke the github pages and related branch altogether.

**Comment by ThisIsMissEm** [comment]
I'm intending on doing that soon, I was just trying to figure out why I had a regression in the docs build on Vercel (Kim had noticed it when I was discussing it with her)

However, since it looks like (at present) it's a bug only in places where we do not have a repo url I'll finish rebasing and merging this soon.

> On 23. Feb 2022, at 19:10, Matthieu Bosquet ***@***.***> wrote:
> 
> ﻿
> @kay-kim OK, thank you for the precision.
> 
> Then unless @ajacksified you have more information, I would suggest we nuke the github pages and related branch altogether.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you were mentioned.

-- 
This e-mail, and any attachments thereto, is intended only for use by the 
addressee(s) named herein and may contain legally privileged, confidential 
and/or proprietary information. If you are not the intended recipient of 
this e-mail (or the person responsible for delivering this document to the 
intended recipient), please do not disseminate, distribute, print or copy 
this e-mail, or any attachment thereto. If you have received this e-mail in 
error, please respond to the individual sending the message, and 
permanently delete the email.

File: docs/api/conf.py (hunk 1 above)
Comment (ThisIsMissEm) [reply docs/api/conf.py:72]: Debugging, was trying to figure out the links issue

**Comment by ajacksified** [comment]
Yup, let's nuke gh-pages

**Review by NSeydoux — APPROVED** [review]

**Comment by ThisIsMissEm** [comment]
Merging without CI passing for E2E tests due to known issue.

**Pull request merged**